backend/src/file/file.service.ts

```diff
@@ -0,0 +1,39 @@
+import { Inject, Injectable } from '@nestjs/common';
+import { MongoGridFS } from 'mongo-gridfs';
+import { GridFSBucketReadStream } from 'mongodb';
+import { FileInfoDto } from './@dtos/file.info.dto';
+import { DATABASE_CONNECTION } from '../database/constants/database.constant';
+import { FileNotFound } from '../common/Exception/file-error.exception';
+
+@Injectable()
+export class FileService {
+  private fileModel: MongoGridFS;
+
+  constructor(@Inject(DATABASE_CONNECTION) private readonly db: any) {
+    const connect = this.db.connections[0].db;
+    this.fileModel = new MongoGridFS(connect, 'fs');
+  }
+
+  async readStream(id: string): Promise<GridFSBucketReadStream> {
+    return await this.fileModel.readFileStream(id);
+  }
+
+  async findInfo(id: string): Promise<FileInfoDto> {
+    try {
+      const result = await this.fileModel.findById(id);
+      return {
+        filename: result.filename,
+        length: result.length,
+        chunkSize: result.chunkSize,
+        md5: result.md5,
+        contentType: result.contentType,
+      };
+    } catch (err) {
+      throw new FileNotFound();
+    }
```

Comment:
bad catch error, it will not throw `FileNotFound()`
add this
```ts
if(!result) throw new FileNotFound();
```
instead and there no need `try/catch` in `async findInfo` scope

Reply:
I try to use 

```ts
if(!result) throw new FileNotFound();
```
but it doesn't throw FileNotFound(), it throws an error of internal server error (500)

Reply:
maybe 
```ts
this.fileModel.findById(id);
```
it already throw from inner, I'm not already confirm with that but if u have been try
I'll approve it